README.md

```diff
@@ -230,6 +230,41 @@ import "github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
 import "github.com/Azure/azure-sdk-for-go/profiles/preview/compute/mgmt/compute"
 ```
 
+## Inspecting and Debugging
+
+All clients implement some handy hooks to help inspect the underlying requests being made to Azure.
+
+- `RequestInspector`: View and manipulate the go `http.Request` before it's sent
+- `ResponseInspector`: View the `http.Response` received
+
+Here is an example of how these can be used with `net/http/httputil` to see requests and responses.
+
+```go
+vnetClient := network.NewVirtualNetworksClient("<subscriptionID>")
+vnetClient.RequestInspector = LogRequest()
+vnetClient.ResponseInspector = LogResponse()
+
+func LogRequest() autorest.PrepareDecorator {
+	return func(p autorest.Preparer) autorest.Preparer {
+		return autorest.PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			dump, _ := httputil.DumpRequestOut(r, true)
```

Comment:
You need to call the passed in preparer first so that the chain is executed in declared order.  See [here](https://github.com/Azure/go-autorest/blob/master/autorest/preparer.go#L104) for an example.